server/src/testFixtures/java/org/elasticsearch/threadpool/TestThreadPool.java

```diff
@@ -38,11 +38,76 @@ public static void dumpThreads() {
         Map<Thread, StackTraceElement[]> allStackTraces = Thread.getAllStackTraces();
         for (var entry : allStackTraces.entrySet()) {
             Thread thread = entry.getKey();
+            if (thread == Thread.currentThread()) {
```

Comment:
sorry, I don't get this check, could you explain?

Reply:
Done